OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/FullscreenDialogFragment.kt

```diff
@@ -0,0 +1,136 @@
+package org.edx.mobile.view.dialog
+
+import android.os.Bundle
+import android.text.Spannable
+import android.text.SpannableStringBuilder
+import android.text.style.ForegroundColorSpan
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.annotation.StringRes
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.viewModels
+import dagger.hilt.android.AndroidEntryPoint
+import org.edx.mobile.R
+import org.edx.mobile.core.IEdxEnvironment
+import org.edx.mobile.databinding.DialogFullscreenLoaderBinding
+import org.edx.mobile.http.HttpStatus
+import org.edx.mobile.util.InAppPurchasesException
+import org.edx.mobile.util.NonNullObserver
+import org.edx.mobile.viewModel.InAppPurchasesViewModel
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class FullscreenLoaderDialogFragment : DialogFragment() {
+
+    @Inject
+    lateinit var environment: IEdxEnvironment
+
+    private lateinit var binding: DialogFullscreenLoaderBinding
+
+    private val iapViewModel: InAppPurchasesViewModel
+            by viewModels(ownerProducer = { requireActivity() })
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(
+            STYLE_NORMAL,
+            R.style.AppTheme_NoActionBar
+        )
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        binding = DialogFullscreenLoaderBinding.inflate(inflater)
+        isCancelable = false
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, args: Bundle?) {
+        super.onViewCreated(view, args)
+        intiViews()
+        initObservers()
+        if (iapViewModel.isVerificationPending) {
```

Comment:
Is this check restricted to only call execute API for once? even after rotation?

Reply:
Yes, this is so that execute API gets called once in any case. The `retry` functionality of `execute` API will be done as part of LEARNER-8813.